test/sign_in_users.spec.js

```diff
@@ -0,0 +1,132 @@
+const supertest = require('supertest');
+const jwt = require('jwt-simple');
+const app = require('../app');
+const userService = require('../app/services/users');
+const errorsCatalog = require('../app/schemas/errors_catalog');
+const errors = require('../app/errors');
+
+const request = supertest(app);
+
+const postUser = (endpoint, data) =>
+  request
+    .post(endpoint)
+    .send(data)
+    .set('Accept', 'application/json');
+
+describe('If POST /users/sessions', () => {
+  describe('When user email is not valid', () => {
+    let response = {};
+    let userFound = {};
+    const userInvalid = {
+      email: 'user@wolo.com.ar',
+      password: 'hola1234'
+    };
+
+    beforeAll(async () => {
+      response = await postUser('/users/sessions', userInvalid);
+      userFound = await userService.emailExists(userInvalid.email);
+    });
+
+    it('Checks error structure given by middleware', () =>
+      expect(Object.keys(response.body)).toEqual(['message', 'internal_code']));
+
+    it('Receive status 400', () => expect(response.status).toBe(400));
+
+    it(`Receive an ${errors.INVALID_PARAMS_ERROR} code`, () =>
+      expect(response.body.internal_code).toEqual(errors.INVALID_PARAMS_ERROR));
+
+    it(`Receive an '${errorsCatalog.EMAIL_ERROR}' message`, () =>
+      expect(response.body.message[0]).toBe(errorsCatalog.EMAIL_ERROR));
+
+    it('The user is not stored in the DB.', () => expect(userFound).toBeFalsy());
+  });
+
+  describe('When user is not registered', () => {
+    let response = {};
+    let userFound = {};
+    const user = {
+      email: 'user@wolox.com.ar',
+      password: 'hola1234'
+    };
+
+    beforeAll(async () => {
+      response = await postUser('/users/sessions', user);
+      userFound = await userService.emailExists(user.email);
+    });
+
+    it('Checks error structure given by middleware', () =>
+      expect(Object.keys(response.body)).toEqual(['message', 'internal_code']));
+
+    it('Receive status 401', () => expect(response.status).toBe(401));
+
+    it(`Receive an ${errors.CREDENTIALS_ERROR} code`, () =>
+      expect(response.body.internal_code).toEqual(errors.CREDENTIALS_ERROR));
+
+    it(`Receive an '${errorsCatalog.CREDENTIALS_ERROR}' message`, () =>
+      expect(response.body.message).toBe(errorsCatalog.CREDENTIALS_ERROR));
+
+    it('The user is not stored in the DB.', () => expect(userFound).toBeFalsy());
+  });
+
+  describe('When user is registered', () => {
+    let response = {};
+    let userFound = {};
+    const userRegister = {
+      email: 'user@wolox.com.ar',
+      password: 'hola1234',
+      name: 'Martin',
+      last_name: 'Acosta'
+    };
+
+    describe('When password not match', () => {
+      const userLogin = {
+        email: 'user@wolox.com.ar',
+        password: 'hola12345'
+      };
+
+      beforeAll(async () => {
+        await postUser('/users', userRegister);
```

Comment:
You can create data using factory girl.
For example, in factory folder you can add file users.js with this logic:
```.js
const { factory } = require('factory-girl');

const { factoryWithCustomizedValue } = require('./factory_by_models');

const modelName = 'User';

factoryWithCustomizedValue(modelName, deletedAt, null);

module.exports = {
  create: userData => factory.create(modelName, userData)
};
```

And in this file requires:
```.js
const { create: createUser } = require('../../factory/users');
```

and finally: await createUser(userLogin);
(Before you should encrypt the password)